### PR TITLE
Program:GCI - Fixed back button handling

### LINKED ISSUE #375: Back Functionality not working properly
Currently, the user can go back to previous question in the game and change its answer. 
To be precise, the user can keep on pressing back button and go back to already visited scenario as well.

This bug is happening after some recent code changes in the codebase. @aanchal07 Can I work on this?

### PR DESCRIPTION
**Pull request by akshaynexus**
### Description
Fixed the improper handling of the back button .

Fixes #375 , #503 

### Type of Change:
**Delete irrelevant options.**

- Code

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?
This has been tested by running the project on my personal android device.

### Checklist:
- [X] My PR follows the style guidelines of this project
- [X] I have performed a self-review of my own code or materials
- [X] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas

**Code/Quality Assurance Only**
- [X] My changes generate no new warnings

### CODE REVIEW

File: PowerUp/app/src/main/java/powerup/systers/com/GameOverActivity.java [hunk 1]
```diff
@@ -31,4 +31,19 @@ public void onClick(View v) {
...
+        Intent intent = new Intent(this, StartActivity.class);
+        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        startActivity(intent);
+        finish();
+    }
+	
+    @Override
+    public void onBackPressed() {
+        super.onBackPressed();
+        goToMap()
```
Comment (Rimjhim28) [inline PowerUp/app/src/main/java/powerup/systers/com/GameOverActivity.java]: Missing ;(semi colon). Syntax Error.

File: PowerUp/app/src/main/java/powerup/systers/com/GameActivity.java [hunk 2]
```diff
@@ -267,4 +267,19 @@ public DatabaseHandler getmDbHandler() {
...
+        Intent intent = new Intent(this, MapActivity.class);
+        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        startActivity(intent);
+        finish();
+    }
+	
+    @Override
+    public void onBackPressed() {
+        super.onBackPressed();
+        goToMap()
```
Comment (Rimjhim28) [inline PowerUp/app/src/main/java/powerup/systers/com/GameActivity.java]: Missing ;(semi colon). Syntax Error

File: PowerUp/app/src/main/java/powerup/systers/com/StoreActivity.java [hunk 3]
```diff
@@ -380,11 +380,21 @@ public void setmDbHandler(DatabaseHandler mDbHandler) {
...
+        startActivity(intent);
+        finish();
+    }
+	
     @Override
     public void onBackPressed() {
-        finish();
         super.onBackPressed();
-    }
+        goToMap()
```
Comment (Rimjhim28) [inline PowerUp/app/src/main/java/powerup/systers/com/StoreActivity.java]: Missing ;(semi colon). Syntax Error

File: PowerUp/app/src/main/java/powerup/systers/com/minesweeper/MinesweeperGameActivity.java [hunk 4]
```diff
@@ -235,6 +235,21 @@ public void continuePressed(View view) {
...
+        Intent intent = new Intent(this, MapActivity.class);
+        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        startActivity(intent);
+        finish();
+    }
+	
+    @Override
+    public void onBackPressed() {
+        super.onBackPressed();
+        goToMap()
```
Comment (Rimjhim28) [inline PowerUp/app/src/main/java/powerup/systers/com/minesweeper/MinesweeperGameActivity.java]: Missing ;(semi colon). Syntax Error

File: PowerUp/app/src/main/java/powerup/systers/com/minesweeper/MinesweeperTutorials.java [hunk 5]
```diff
@@ -34,5 +34,21 @@ public void onClick(View v) {
...
+        Intent intent = new Intent(this, MapActivity.class);
+        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        startActivity(intent);
+        finish();
+    }
+	
+    @Override
+    public void onBackPressed() {
+        super.onBackPressed();
+        goToMap()
```
Comment (Rimjhim28) [inline PowerUp/app/src/main/java/powerup/systers/com/minesweeper/MinesweeperTutorials.java]: Missing ;(semi colon). Syntax Error

File: PowerUp/app/src/main/java/powerup/systers/com/minesweeper/ProsAndConsActivity.java [hunk 6]
```diff
@@ -49,5 +49,20 @@ public void continuePressedProsAndCons(View v) {
...
+        Intent intent = new Intent(this, MapActivity.class);
+        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        startActivity(intent);
+        finish();
+    }
+	
+    @Override
+    public void onBackPressed() {
+        super.onBackPressed();
+        goToMap()
```
Comment (Rimjhim28) [inline PowerUp/app/src/main/java/powerup/systers/com/minesweeper/ProsAndConsActivity.java]: Missing ;(semi colon). Syntax Error

File: PowerUp/app/src/main/java/powerup/systers/com/sink_to_swim_game/SinkToSwimEndActivity.java [hunk 7]
```diff
@@ -33,4 +33,19 @@ public void continuePressed(View view){
...
+        Intent intent = new Intent(this, MapActivity.class);
+        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        startActivity(intent);
+        finish();
+    }
+	
+    @Override
+    public void onBackPressed() {
+        super.onBackPressed();
+        goToMap()
```
Comment (Rimjhim28) [inline PowerUp/app/src/main/java/powerup/systers/com/sink_to_swim_game/SinkToSwimEndActivity.java]: Missing ;(semi colon). Syntax Error

File: PowerUp/app/src/main/java/powerup/systers/com/sink_to_swim_game/SinkToSwimGame.java [hunk 8]
```diff
@@ -287,4 +287,19 @@ public void onPause() {
...
+        Intent intent = new Intent(this, MapActivity.class);
+        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        startActivity(intent);
+        finish();
+    }
+	
+    @Override
+    public void onBackPressed() {
+        super.onBackPressed();
+        goToMap()
```
Comment (Rimjhim28) [inline PowerUp/app/src/main/java/powerup/systers/com/sink_to_swim_game/SinkToSwimGame.java]: Missing ;(semi colon). Syntax Error

File: PowerUp/app/src/main/java/powerup/systers/com/sink_to_swim_game/SinkToSwimTutorials.java [hunk 9]
```diff
@@ -49,5 +49,21 @@ public void onClick(View v) {
...
+        Intent intent = new Intent(this, MapActivity.class);
+        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        startActivity(intent);
+        finish();
+    }
+	
+    @Override
+    public void onBackPressed() {
+        super.onBackPressed();
+        goToMap()
```
Comment (Rimjhim28) [inline PowerUp/app/src/main/java/powerup/systers/com/sink_to_swim_game/SinkToSwimTutorials.java]: Missing ;(semi colon). Syntax Error

File: PowerUp/app/src/main/java/powerup/systers/com/vocab_match_game/VocabMatchEndActivity.java [hunk 10]
```diff
@@ -36,4 +36,20 @@ public void continuePressed(View view){
...
+        Intent intent = new Intent(this, MapActivity.class);
+        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        startActivity(intent);
+        finish();
+    }
+	
+    @Override
+    public void onBackPressed() {
+        super.onBackPressed();
+        goToMap()
```
Comment (Rimjhim28) [inline PowerUp/app/src/main/java/powerup/systers/com/vocab_match_game/VocabMatchEndActivity.java]: Missing ;(semi colon). Syntax Error

File: PowerUp/app/src/main/java/powerup/systers/com/vocab_match_game/VocabMatchGameActivity.java [hunk 11]
```diff
@@ -217,4 +217,20 @@ public boolean onTouch(View v, MotionEvent motionEvent) {
...
+        Intent intent = new Intent(this, MapActivity.class);
+        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        startActivity(intent);
+        finish();
+    }
+	
+    @Override
+    public void onBackPressed() {
+        super.onBackPressed();
+        goToMap()
```
Comment (Rimjhim28) [inline PowerUp/app/src/main/java/powerup/systers/com/vocab_match_game/VocabMatchGameActivity.java]: Missing ;(semi colon). Syntax Error

File: PowerUp/app/src/main/java/powerup/systers/com/vocab_match_game/VocabMatchTutorials.java [hunk 12]
```diff
@@ -34,4 +34,20 @@ public void onClick(View v) {
...
+        Intent intent = new Intent(this, MapActivity.class);
+        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        startActivity(intent);
+        finish();
+    }
+	
+    @Override
+    public void onBackPressed() {
+        super.onBackPressed();
+        goToMap()
```
Comment (Rimjhim28) [inline PowerUp/app/src/main/java/powerup/systers/com/vocab_match_game/VocabMatchTutorials.java]: Missing ;(semi colon). Syntax Error

**Comment by akshaynexus** [comment]
Oh ok

**Pull request closed without merging**